glidein_start.sh

```diff
@@ -198,8 +198,23 @@ export PATH=$PATH:$_condor_SBIN:$PWD/glideinExec/bin
 export LD_LIBRARY_PATH=$_condor_LIB:$_condor_LIB/condor:$LD_LIBRARY_PATH
 
 # run condor
-exec glideinExec/sbin/condor_master -dyn -f -r 1200
-
-# clean up after ourselves
-cd ..
-rm -rf glidein
+trap 'kill -TERM $PID; if [ -n $PID_LOG_SHIPPER ]; then kill -TERM $PID_LOG_SHIPPER; fi' SIGTERM SIGKILL
+glideinExec/sbin/condor_master -dyn -f -r 1200 &
+PID=$!
+
+# starting log_shipper
+if [ -n $PRESIGNED_PUT_URL ]
+  then
+    ../log_shipper.sh ${PRESIGNED_PUT_URL} &
+    PID_LOG_SHIPPER=$!
+fi
+
+wait $PID
+trap - SIGTERM SIGKILL
+wait $PID
```

Comment:
Is it guaranteed that the `log_shipper.sh` is dead here, or do we need to wait on it too?

Reply:
I decided to skip waiting on it and let the glidein_start.sh script run one last upload at the end of the script.  I figured this gave us the best chances for one last good upload